engine/api/src/main/java/io/deephaven/engine/table/ColumnDefinition.java

```diff
@@ -424,8 +417,11 @@ public boolean isDirect() {
      * @return True if the ColumnDefinition defines a column whose data is compatible with this ColumnDefinition.
      */
     public boolean isCompatible(ColumnDefinition<?> other) {
+        if (this == other) {
+            return true;
+        }
         return this.name.equals(other.name)
-                && this.dataType.equals(other.dataType)
+                && this.dataType == other.dataType
```

Comment:
This is arguably safer. I can't find anything in the JLS to support this, but to the best of my knowledge `equals()` will return `true` even when two classes were loaded with different `ClassLoaders`, but `==` will not. In this case, you prefer to find these classes unequal; otherwise you may run into issues later (e.g. `ArrayStoreException`).